graphql_compiler/schema_transformation/utils.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from graphql.language.ast import NamedType
+from graphql.language.visitor import Visitor, visit
+from graphql.type.definition import GraphQLScalarType
+
+
+class SchemaTransformError(Exception):
+    """Parent of specific error classes."""
+
+
+class SchemaStructureError(SchemaTransformError):
+    """Raised if an input schema's structure is illegal.
+
+    This may happen if an AST cannot be built into a schema, if the schema contains disallowed
+    components, or if the schema contains some field of the query type that is named differently
+    from the type it queries.
+    """
+
+
+class SchemaNameConflictError(SchemaTransformError):
+    """Raised when renaming types or fields cause name conflicts."""
+
+
+def get_query_type_name(schema):
+    """Get the name of the query type of the input schema.
+
+    Args:
+        schema: GraphQLSchema
+
+    Returns:
+        str, name of the query type (e.g. RootSchemaQuery)
+    """
+    return schema.get_query_type().name
+
+
+def get_scalar_names(schema):
+    """Get names of all scalars used in the input schema.
+
+    Includes all user defined scalars, as well as any builtin scalars used in the schema; excludes
+    builtin scalars not used in the schema.
+
+    Returns:
+        Set[str], set of names of scalars used in the schema
+    """
+    type_map = schema.get_type_map()
+    scalars = {
+        type_name
+        for type_name in type_map
```

Comment:
micro nit: if you make this `for type_name, type_object in six.iteritems(type_map)` then you can avoid the `type_map[type_name]` lookup on the line below.